hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/scm/pipeline/TestPipelineManagerMXBean.java

```diff
@@ -83,14 +89,46 @@ private void verifyEquals(TabularData actualData,
       assertEquals(2, cds.values().size());
       Iterator<?> it = cds.values().iterator();
       String key = it.next().toString();
-      String value = it.next().toString();
-      long num = Long.parseLong(value);
       assertTrue(expectedData.containsKey(key));
-      assertEquals(expectedData.remove(key).longValue(), num);
+      // Wait before all pipelines becomes stable in their respective states.
+      GenericTestUtils.waitFor(() -> {
+        long actualCountForPipelineState =
+            getActualCountForPipelineState(key, bean);
+        return expectedData.remove(key).longValue() ==
+            actualCountForPipelineState;
+      }, 500, 3000);
```

Comment:
`waitFor` may repeatedly call this block until it returns `true` (within the given time limit).  But `expectedData.remove(key)` gives `null` on the second and later calls.

```suggestion
      final long expected = expectedData.remove(key).longValue();
      // Wait before all pipelines becomes stable in their respective states.
      GenericTestUtils.waitFor(() -> 
          getActualCountForPipelineState(key, bean) == expected,
          500, 3000);
```

Given that it passed hundreds of times anyway, maybe we could use `assertEquals` instead of `waitFor`.